### PR TITLE
Port to fastapi

### LINKED ISSUE #1293: rewrite DMT API to FastAPI

### PR DESCRIPTION
**Pull request by KristianKjerstad**
## What does this pull request change?
* change flask to port API in DMT API

## Issues related to this change
closes #1293

### CODE REVIEW

File: api/src/app.py [hunk 1]
```diff
@@ -5,49 +5,41 @@
...
 from config import config
-from controllers import blueprints, entity, system, jobs
+
 from services.dmss import dmss_api
 from use_case.import_package import import_package_tree, package_tree_from_zip
 from utils.create_application_utils import zip_all
 from utils.logging import logger
+from store_headers_middleware import StoreHeadersMiddleware
 
+prefix = "/api"
```
Comment (netr0m) [inline api/src/app.py:24]:
```suggestion
prefix = "/v1/api"
```
Perhaps we should use version prefixing here as well? (as we already do in DMSS)

Comment (netr0m) [inline api/src/app.py:24]:
```suggestion
prefix = "/api/v1"
```
Perhaps we should prefix with the version, as we do in DMSS?

Comment (netr0m) [reply api/src/app.py:24]: https://github.com/equinor/data-modelling-storage-service/blob/master/src/app.py#L21-L23

File: api/src/restful/use_case.py [hunk 2]
```diff
@@ -1,38 +1,62 @@
...
+    status: int,
+) -> JSONResponse:
+    types = {
+        400: "PARAMETERS_ERROR",
+        401: "UNAUTHORIZED",
+        403: "FORBIDDEN",
+        404: "RESOURCE_ERROR",
+        422: "UNPROCESSABLE_ENTITY",
+        500: "SYSTEM_ERROR",
+    }
```
Comment (netr0m) [inline api/src/restful/use_case.py:26]: Not important, but perhaps we could reuse the `STATUS_CODES` from enums.py somehow?

**Review by netr0m — APPROVED** [review]
LGTM! A couple of minor remarks to consider

File: api/src/controllers/jobs.py [hunk 3]
```diff
@@ -1,40 +1,39 @@
...
 
 if config.JOB_SERVICE_ENABLED:
 
-    @blueprint.route("/api/job/<path:job_id>", methods=["POST"])
+    @router.post("/{job_id:path}", operation_id="start_job")
     def start(job_id: str):
         use_case = StartJobUseCase()
-        response = use_case.execute({"job_id": job_id})
-        return Response(json.dumps(response.value), mimetype="application/json", status=STATUS_CODES[response.type])
+        return use_case.execute({"job_id": job_id})
```
Comment (eoaksnes) [inline api/src/controllers/jobs.py:19]: We should align the use case logic with the template. We can do that in a separate PR.

**Review by eoaksnes — APPROVED** [review]

File: api/src/controllers/jobs.py (hunk 3 above)
Comment (KristianKjerstad) [reply api/src/controllers/jobs.py:19]: good idea, will create a new issue for this

File: api/src/restful/use_case.py (hunk 2 above)
Comment (KristianKjerstad) [reply api/src/restful/use_case.py:26]: good point, it actually looks like this types dict is not needed. we can use the status dict from starlette, which gives more consistency.

This means we can delete the STATUS_CODES in enums.py also
